server/container/guice/blob/deduplication-gc/src/main/java/org/apache/james/modules/blobstore/BlobDeduplicationGCModule.java

```diff
@@ -68,8 +69,15 @@ protected void configure() {
 
     @Singleton
     @Provides
-    public GenerationAwareBlobId.Factory generationAwareBlobIdFactory(Clock clock, PlainBlobId.Factory delegate, GenerationAwareBlobId.Configuration configuration) {
-        return new GenerationAwareBlobId.Factory(clock, delegate, configuration);
+    public BlobId.Factory generationAwareBlobIdFactory(Clock clock, PlainBlobId.Factory delegate, GenerationAwareBlobId.Configuration configuration) {
+            String property = System.getProperty("james.s3.minio.compatibility.mode");
```

Comment:
Should we add a doc in `jvm.adoc`?

Reply:
Yes and likely a one line into blob store configuration